src/swap/components/SwapSettings.tsx

```diff
@@ -0,0 +1,37 @@
+import { isValidElement, useMemo } from 'react';
+import { SwapSettingsSvg } from '../../internal/svg/swapSettings';
+import { cn } from '../../styles/theme';
+import type { SwapSettingsReact } from '../types';
+
+export function SwapSettings({
+  className,
+  icon,
+  title = 'Auto',
+}: SwapSettingsReact) {
+  const iconSvg = useMemo(() => {
```

Comment:
We probably want to re-use this https://github.com/coinbase/onchainkit/blob/704e16081acf8e03a821e06e25e33fe4a68fc63c/src/wallet/hooks/useIcon.tsx#L2, and elevate the `useIcon` utilities in a new folder.

```
src/hooks/useIcon.ts
```

Reply:
Update to reuse the `useIcon` hook